src/pal/pal_posix.h

```diff
@@ -30,6 +30,64 @@ namespace snmalloc
   template<class OS>
   class PALPOSIX
   {
+    /**
+     * Helper class to access the `default_mmap_flags` field of `OS` if one
+     * exists or a default value if not.  This provides the default version,
+     * which is used if `OS::default_mmap_flags` does not exist.
+     */
+    template<typename T, typename = int>
+    struct DefaultMMAPFlags
+    {
+      /**
+       * If `OS::default_mmap_flags` does not exist, use 0.  This value is
+       * or'd with the other mmap flags and so a value of 0 is a no-op.
+       */
+      static const int flags = 0;
+    };
+
+    /**
+     * Helper class to access the `default_mmap_flags` field of `OS` if one
+     * exists or a default value if not.  This provides the version that
+     * accesses the field, allowing other PALs to provide extra arguments to
+     * the `mmap` calls used here.
+     */
+    template<typename T>
+    struct DefaultMMAPFlags<T, decltype((void)T::default_mmap_flags, 0)>
+    {
+      static const int fd = T::default_mmap_flags;
```

Comment:
```suggestion
      static const int flags = T::default_mmap_flags;
```

Fix copy-and-paste error.